server/routes/updateGoal/updateGoalController.ts

```diff
@@ -43,6 +43,22 @@ export default class UpdateGoalController {
     const updateGoalForm: UpdateGoalForm = { ...req.body }
     req.session.updateGoalForm = updateGoalForm
 
+    // Remove the desired step on the action delete step
+    if (updateGoalForm.action && updateGoalForm.action.startsWith('delete-step-')) {
```

Comment:
Check that there is a updateGoalForm.action when using `startsWith` due to test error

```
UpdateGoalController › submitUpdateGoalForm › should redirect to update goal form given validation fails

    TypeError: Cannot read properties of undefined (reading 'startsWith')
```